uropa/visualization.py

```diff
@@ -1,77 +1,125 @@
+
+    return distPlot
 
 
-def peak_count_plot(table, var, kind):
+def peak_count_plot(table, var="feature", peak_type="exon", group_by=["peak_chr", "peak_start", "peak_end", "peak_strand"], stacked=False, color_by=None, title=None, output=None, dpi=300):
     """
     Count and plot occurence of selected variable by peak.
 
     Parameters
     ----------
     table : pd.DataFrame
-        Pandas dataframe containing the data.
-    var : <datatype>, <default value>
-        <param description>
-    kind : <datatype>, <default value>
-        <param description>
-    TODO add more parameters
+        Pandas dataframe containing the data
+    var : string, default "feature"
+        Column which table is filtered by
+    peak_type : string, default "exon"
+        Value of the selected column (var)
+    group_by : list of string, default ["peak_chr", "peak_start", "peak_end", "peak_strand"]
+        Columns which table is grouped by
+    stacked : bool, default False
+        If true, display a stacked barplot (only works if color_by is not None), else display an unstacked barplot
+    color_by : string, default None
+        Column by whose values the plot should be colored
+    title : string, default None
+        Title of the plot
+    output : string, default None
+        Path where the plot should be saved
+    dpi : Int, default 300
+        Resolution of the plot
 
     Returns
     -------
-    <datatype> :
-        <return description>
-    TODO should return the plotting object
+    matplotlib.axes._subplots.AxesSubplot :
+        Plot object for further processing
     """
-    pass
+
+    sns.set_style("darkgrid")
+
+    # Filter DataFrame by column var and peak_type
+    df = table[table[var] == peak_type]
+
+    if color_by:
+        group_by.append(color_by)
+        # Count the number of equal columns which belong to the group_by list and add the column to the DataFrame
+        df = df.groupby(group_by).size().to_frame(peak_type)
+        # Count the corresponding numbers of the color_by values and add the column to the DataFrame ("count")
+        df = df.groupby([peak_type, color_by]).size().to_frame(
+            "count").reset_index()
+
+        # Generate stacked barplot
+        if stacked:
+            pcPlot = df.pivot(index=peak_type, columns=color_by, values="count").plot(
+                kind="bar", stacked=True, rot=0)
+            pcPlot.set_ylabel("count")
+        # Generate barplot
+        else:
+            pcPlot = sns.barplot(data=df, x=peak_type, y="count", hue=color_by)
+            sns.move_legend(pcPlot, "upper right")
+    else:
+        # Count the number of equal columns which belong to the group_by list
+        df = df.groupby(group_by).size()
+        # Generate histogram
+        pcPlot = sns.histplot(data=df, discrete=True)
+        pcPlot.set(xlabel=peak_type, ylabel="count")
+
+    if title:
+        pcPlot.set(title=title)
+
+    if output:
+        plt.savefig(output, dpi=dpi)

```

Comment:
`dpi` has only effect on the saved image. Please change that so it will always apply.